### PR TITLE
Add calculationID to the entity -> DTO

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: src/main/resources/db/migration/V2_0__RECREATE_TIER_CALCULATION.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS TIER_CALCULATION;
```
Comment (nickmcmahon01) [inline src/main/resources/db/migration/V2_0__RECREATE_TIER_CALCULATION.sql:1]: obviously here we'll lose data,
We can clear all the databases before we go live and consolidate these scripts.

Comment (markreesmoj) [reply src/main/resources/db/migration/V2_0__RECREATE_TIER_CALCULATION.sql:1]: If we get this in prod before testing starts it will be fine I think.

**Comment by markreesmoj** [comment]
Maybe some of the lower level tests can be replaced by tests of the endpoint, once written?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt [hunk 2]
```diff
@@ -28,9 +29,16 @@ class TierCalculationController(private val tierCalculationService: TierCalculat
       ApiResponse(code = 404, message = "Result Not Found")
     ]
   )
+
   @GetMapping("crn/{crn}/tier")
-  fun getNotifications(@PathVariable(required = true) crn: String): ResponseEntity<TierDto> {
-    return ResponseEntity.ok(tierCalculationService.getTierByCrn(crn))
+  fun getLatestTierCalculation(@PathVariable(required = true) crn: String): ResponseEntity<TierDto> {
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt:34]: Do we need this endpoint now?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt [hunk 3]
```diff
@@ -28,9 +29,16 @@ class TierCalculationController(private val tierCalculationService: TierCalculat
...
+    return ResponseEntity.ok(tierCalculationService.getLatestTierByCrn(crn))
+      ?: throw EntityNotFoundException("Tier Result Not Found for $crn")
+  }
+
+  @GetMapping("crn/{crn}/tier/{calculationId}")
+  fun getTierCalculationById(@PathVariable(required = true) crn: String, @PathVariable(required = true) calculationId: UUID): ResponseEntity<TierDto> {
+    return ResponseEntity.ok(tierCalculationService.getTierByCalculationId(crn, calculationId))
       ?: throw EntityNotFoundException("Tier Result Not Found for $crn")
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt:42]: should we include the UUID in this message?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt [hunk 4]
```diff
@@ -57,8 +58,55 @@ class TierCalculationRepositoryTest(
...
+      val calculationId = UUID.randomUUID()
+      val created = LocalDateTime.now()
+      val firstTierCalculation = TierCalculationEntity(crn = crn, created = created, data = data, uuid = calculationId)
+
+      repository.save(firstTierCalculation)
+
+      val calculation = repository.findByCrnAndUuid(crn, calculationId)
+      assertThat(calculation).isNotNull
+      assertThat(calculation?.crn).isEqualTo(firstTierCalculation.crn)
+      assertThat(calculation?.created).isEqualToIgnoringNanos(firstTierCalculation.created)
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt:91]: 😮 
What a remarkable matcher

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt [hunk 5]
```diff
@@ -57,8 +58,55 @@ class TierCalculationRepositoryTest(
     fun `Should return latest calculation when multiple`() {
 
       val created = LocalDateTime.now()
-      val firstTierCalculation = TierCalculationEntity(crn = crn, created = created, data = data)
-      val secondTierCalculation = TierCalculationEntity(crn = "$crn 2", created = created.minusSeconds(1), data = data)
+      val firstTierCalculation = TierCalculationEntity(crn = crn, created = created, data = data, uuid = UUID.randomUUID())
+      val secondTierCalculation = TierCalculationEntity(crn = "$crn 2", created = created.minusSeconds(1), data = data, uuid = UUID.randomUUID())
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt:62]: I might be misreading this but it looks like the repository has calculations with different CRNs so findFirst will always return the one which matches the crn

**Review by markreesmoj — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt (hunk 2 above)
Comment (nickmcmahon01) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt:34]: I think we need to clarify this. I can't think of a specific use case, but we master the data now so need to make it available to other clients?

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/TierCalculationController.kt:34]: Feels like we are doing half a job here. If there is a need for it we need to get the access restrictions and documentation etc right, and test it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt (hunk 5 above)
Comment (nickmcmahon01) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepositoryTest.kt:62]: That's a really good point, I'll check the git blame. 🤡

**Pull request merged**